modules/tappxBidAdapter.js

```diff
@@ -0,0 +1,284 @@
+'use strict';
+
+import * as utils from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { config } from '../src/config.js';
+
+const BIDDER_CODE = 'tappx';
+var HOST;
+var hostDomain;
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  isBidRequestValid: function(bid) {
+    if ((bid.params == null) || (bid.params.endpoint == null) || (bid.params.tappxkey == null)) {
+      utils.logWarn(`[TAPPX]: The Tappx endpoint and Tappxkey are mandatory. ${JSON.stringify(bid)}`);
+      return false;
+    }
+    return true;
+  },
+
+  /**
+   * Takes an array of valid bid requests, all of which are guaranteed to have passed the isBidRequestValid() test.
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    // Host definition
+    HOST = getParameters(validBidRequests, 'params.host');
+    hostDomain = HOST.split('/', 1)[0];
+
+    const ENDPOINT = getParameters(validBidRequests, 'params.endpoint');
+    const TAPPXKEY = getParameters(validBidRequests, 'params.tappxkey');
+    const BIDFLOOR = getParameters(validBidRequests, 'params.bidfloor');
```

Comment:
I'd like to bring your attention to Price Floors Module in PBJS, https://docs.prebid.org/dev-docs/modules/floors.html, this module provides a systematic way for publishers to configure and adapters to read price floor data. Did you consider reading price floor data from this module (when available)?

Reply:
Initially, we prefer to design the adapter as simple as possible, without modules and similar to our OpenRTB requests. We are designing the video feature and maybe we will consider adding the floor module with that function.